tests/Doctrine/Tests/ORM/Internal/Cache/LazyPropertyMapTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LazyMapTest;
+
+use Doctrine\ORM\Internal\Hydration\Cache\LazyPropertyMap;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+
+/**
+ * @link https://github.com/Ocramius/LazyMap/blob/1.0.0/tests/LazyMapTest/CallbackLazyMapTest.php
+ *
+ * @covers \Doctrine\ORM\Internal\Hydration\Cache\LazyPropertyMap
+ */
+class LazyPropertyMapTest extends TestCase
+{
+    /**
+     * @var \LazyMap\CallbackLazyMap
+     */
+    protected $lazyMap;
+
+    /**
+     * @var callable|\PHPUnit_Framework_MockObject_MockObject
+     */
+    protected $callback;
+
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $this->callback = $this->getMockBuilder(stdClass::class)->setMethods(['__invoke'])->getMock();
+        $this->lazyMap  = new LazyPropertyMap($this->callback);
+    }
+
+    public function testDirectPropertyAccess()
+    {
+        $count = 0;
+        $this
+            ->callback
+            ->expects(self::exactly(3))
+            ->method('__invoke')
+            ->will(self::returnCallback(function ($name) use (& $count) {
```

Comment:
`willReturnCallback()`?

Reply:
Also, isn't it simpler to not have a mock here? You don't necessarily need to track the number of calls twice.